docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,371 @@
+
+##### MeshTrafficPermission
+
+###### Old ExternalService behaviour
+
+Both MeshTrafficPermission and TrafficPermission only works with egress.
+
+###### New behaviour
+
+It is already implemented on egress for MeshTrafficPermission.
+
+If on the sidecar:
+- could you circumvent this limitation by trying to directly call through egress?
+
+##### MeshLoadBalancingStrategy
+
+###### Old ExternalService behaviour
+
+There is no old behaviour because there was no concept of Endpoints in External Service
+
+###### New behaviour
+
+Only the `loadBalancer` field makes sense in this regard.
+
+If on sidecar:
+- if you don't need/want egress it still makes sense to pick load balancing mechanism
+
+If on egress:
+- makes the same sense on as on sidecar
+
+##### MeshTCPRoute and MeshHTTPRoute
+
+###### Old ExternalService behaviour
+
+It worked for both egress and sidecar.
+
+###### New behaviour
+
+I don't see why we would change that behaviour.
+
+#### For outbound policies on the sidecar, for inbound policies on egress
+
+It naturally fits our targetRef mechanism.
+It does not make sense to apply inbound policies on the sidecar since there is no inbound for it.
+Egress plays a role of a "sidecar" that is inside the mesh, and we can target it's inbound.
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+###### `to`
+
+In `to` section we want to capture all the traffic going `to` a `MeshExternalService.
+
+The situation is pretty straightforward:
+
+```yaml
+kind: MeshTimeout
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: sidecar # just to be explicit that it's on the sidecar? see section 'Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense'
+  to:
+  - targetRef:
+      kind: MeshExternalService
+    default:
+      idleTimeout: 20s
+```
+
+###### `from`
+
+It never makes sense to put `MeshExternalService` in `from` section because traffic never comes from `MeshExternalService`.
+
+###### Top level
+
+We want to have the option to configure inbound policies for `MeshExternalService`.
+
+One option is to use `kind: Mesh`, `proxyType` + `sectionName` just like in [MeshService](https://github.com/kumahq/kuma/blob/7e69b872404251d4e5cd766225fdd1b821cc6c5d/docs/madr/decisions/046-meshservice-policy-matching.md?plain=1#L192):
+
+```yaml
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: egress # when it's egress it's MES name
+    sectionName: httpbin
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        backends:
+          - type: File
+            file:
+              path: /tmp/access.log
+```
+
+Another option is to use `MeshExternalService` as a top level target + `proxyType: egress` + `name`.
+
+```yaml
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: MeshExternalService
+    proxyType: egress
+    name: httpbin
```

Comment:
This a bit depends on https://github.com/kumahq/kuma/pull/10858/files and  https://github.com/kumahq/kuma/pull/10858/files#diff-95c559a0f6155d986d45f294106f1a5a789e99bd0c0c93dae4405724efc76f3cR47-R49 . I guess we could say we're selecting an egress sidecar.